### ISSUE TITLE
Bump Rust toolchain to 1.45

### ISSUE THREAD

**Issue by binarylogic** [timberio/vector#3515]
We need to bump the Rust toolchain to 1.45 to unblock https://github.com/timberio/vector/pull/3514. We've bumped this in the past, but it broke nightly releases because we need to upgrade to LLVM10.

**Comment by tyrken** [comment]
FYI:  The vector nightly builds have been failing for a week or so on Windows, which is slightly annoying as I'd like to switch from my fork to your latest builds (hopefully a release soon), and you only publish if all builds complete successfully.

The cause of the Windows fails looks suspiciously like https://github.com/rust-lang/rust/issues/72470 which is [confirmed](https://github.com/rust-lang/rust/issues/72470#issuecomment-667062608) as being in toolchain 1.44, they attempted to revert afterwards but don't know if its OK in 1.45 or even latest yet...

**Comment by yoshuawuyts** [comment]
> but don't know if its OK in 1.45 or even latest yet...

We're still experiencing this issue at work, and remain pinned to Rust 1.43.0. To my knowledge this has not been addressed in the compiler yet.

**Comment by Hoverbear** [comment]
@tyrken Can you open a separate issue for that? I can try to address it. :)

**Comment by jamtur01** [comment]
@Hoverbear @tyrken That bug is addressed in 1.46 from what I can see from my initial take on this (#3700) but obviously that whole path is dependent on the LLVM upgrade.

**Comment by leebenson** [comment]
Bumping to rustc 1.45 is no longer required in #3514, per moving to async-graphql. I think we can skip this and bump straight to 1.46 once LLVM 10 lands. Closing.
